Fix #110 - issue with beSupersetOf when compare mutable and immutable co...

chipp opened the pull request:
`[subset isKindOfClass:[actual class]]` method doesn't work in cases of when we compare mutable and immutable collection, e.g. NSMutableArray and NSArray.
NSMutableArray is implemented in its subclass __NSArrayM and NSArray is implemented in its subclass __NSArrayI.

In this case isKindOfClass will return NO, because __NSArrayM is subclass of NSArray but not of __NSArrayI.

orta commented:
Thanks a lot @chipp - Any chance we can get a unit test added to prove that it works? :dancers:

chipp commented:
It's looks like someone excluded `EXPMatchers_beSupersetOfTest.m` from test targets, I've added it back and added tests for mutable and immutable collections comparison.

orta commented:
nice! Thanks
